tests/components/librehardwaremonitor/test_config_flow.py

```diff
@@ -0,0 +1,99 @@
+"""Test the LibreHardwareMonitor config flow."""
+
+from unittest.mock import AsyncMock
+
+from librehardwaremonitor_api import (
+    LibreHardwareMonitorConnectionError,
+    LibreHardwareMonitorNoDevicesError,
+)
+
+from homeassistant.components.librehardwaremonitor.const import DOMAIN
+from homeassistant.config_entries import SOURCE_USER
+from homeassistant.const import CONF_HOST, CONF_PORT
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+
+from .conftest import VALID_CONFIG
+
+from tests.common import MockConfigEntry
+
+
+async def test_create_entry(
+    hass: HomeAssistant,
+    mock_setup_entry: AsyncMock,
+    mock_lhm_client: AsyncMock,
+    mock_config_entry: MockConfigEntry,
+) -> None:
+    """Test that a complete config entry is created."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}, data=VALID_CONFIG
+    )
+
+    assert result
+    assert result["type"] is FlowResultType.CREATE_ENTRY
+    assert result["result"].unique_id is None
+
+    mock_config_entry = result["result"]
+    assert (
+        mock_config_entry.title
+        == f"{VALID_CONFIG[CONF_HOST]}:{VALID_CONFIG[CONF_PORT]}"
+    )
+    assert mock_config_entry.data == {
+        CONF_HOST: VALID_CONFIG[CONF_HOST],
+        CONF_PORT: VALID_CONFIG[CONF_PORT],
+    }
+
+    assert mock_setup_entry.call_count == 1
+
+
+async def test_errors_and_flow_recovery(
+    hass: HomeAssistant, mock_setup_entry: AsyncMock, mock_lhm_client: AsyncMock
+) -> None:
+    """Test that errors are shown as expected."""
+    mock_lhm_client.get_data.side_effect = LibreHardwareMonitorConnectionError()
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}, data=VALID_CONFIG
+    )
+
+    assert result
+    assert result["errors"] == {"base": "cannot_connect"}
+    assert result["step_id"] == "user"
+
+    mock_lhm_client.get_data.side_effect = LibreHardwareMonitorNoDevicesError()
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}, data=VALID_CONFIG
+    )
```

Comment:
This is creating a new flow instead of re-using the old one

Reply:
Fixed it.